packages/cyberstorm/src/components/DropDown.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { ReactElement, ReactNode, useState } from "react";
+import styles from "./componentStyles/DropDown.module.css";
+import { Button } from "./Button";
+import { faChevronDown } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+import * as DropdownMenu from "@radix-ui/react-dropdown-menu";
+
+type DropDownProps = {
+  label?: string;
+  icon?: ReactNode;
+  defaultOpen: boolean;
+  dropDownData?: DataItem[];
+  colorScheme?: "default" | "defaultDark" | "primary";
+  defaultValue?: number;
+};
+
+type DataItem = {
+  label: string;
+  value: number;
+  reactElement?: ReactElement;
+};
+
+export const DropDown: React.FC<DropDownProps> = (props) => {
+  const { icon, colorScheme, defaultOpen, dropDownData, defaultValue } = props;
+
+  let initialValue = dropDownData ? dropDownData[0] : null;
+  if (defaultValue && dropDownData && dropDownData[defaultValue]) {
+    initialValue = dropDownData[defaultValue];
+  }
+  const [value, setValue] = useState(initialValue);
+  const dropDownItemElements = mapDropDownData(
+    dropDownData,
+    colorScheme,
+    setValue
+  );
+
+  return (
+    <div className={styles.root}>
+      <DropdownMenu.Root defaultOpen={defaultOpen}>
+        <DropdownMenu.Trigger asChild>
+          <div>
+            <Button
+              colorScheme={colorScheme}
+              rightIcon={icon}
+              label={value ? value.label : ""}
+            />
+          </div>
+        </DropdownMenu.Trigger>
+
+        <DropdownMenu.Portal>
+          <DropdownMenu.Content
+            align={"start"}
+            sideOffset={8}
+            className={`${styles.content} ${getContentStyle(colorScheme)}`}
+          >
+            <DropdownMenu.Group className={styles.group}>
+              {dropDownItemElements}
+            </DropdownMenu.Group>
+          </DropdownMenu.Content>
+        </DropdownMenu.Portal>
+      </DropdownMenu.Root>
+    </div>
+  );
+};
+
+DropDown.defaultProps = {
+  dropDownData: [],
+  defaultOpen: false,
+  colorScheme: "default",
+  defaultValue: 0,
+  icon: (
+    <FontAwesomeIcon
+      fixedWidth={true}
+      icon={faChevronDown}
+      className={"dropDownIcon"}
+    />
+  ),
+};
+
+const getContentStyle = (scheme: DropDownProps["colorScheme"] = "default") => {
+  return {
+    default: styles.content__default,
+    defaultDark: styles.content__defaultDark,
+    primary: styles.content__primary,
+  }[scheme];
+};
+
+const mapDropDownData = (
+  dropDownData: DataItem[] | undefined,
+  colorScheme: "default" | "defaultDark" | "primary" | undefined,
+  setValue: React.Dispatch<React.SetStateAction<DataItem | null>>
+) => {
+  return dropDownData
```

Comment:
Assuming the `reactElement` is typed as `typeof DropDownItem`, we could utilise different things to clean this up into:

```typescript
  const props = { colorScheme } as DropDownItemProps;

  return (dropDownData ?? []).map((dataItem, index) => (
    <DropdownMenu.Item onSelect={() => setValue(dataItem)} key={index} asChild>
      {React.isValidElement(dataItem.reactElement)
        ? React.cloneElement(dataItem.reactElement, props)
        : null}
    </DropdownMenu.Item>
  ));
```

As a plus I think it would have a better type safety. But use your own judgement if any or all of the changes make sense.